### PR TITLE
Expand acceptance tests to verify yum::server and yum::repo interactions

### PR DESCRIPTION
**Pull request by treydock**
This works very much like the existing tests in Vagrant.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,12 @@ matrix:
       env: BEAKER_set="centos-6" BEAKER_PUPPET_COLLECTION="puppet5"
       bundler_args:
       script: bundle exec rake beaker
+    - rvm: 2.4.4
+      sudo: required
+      services: docker
+      env: BEAKER_set="centos-7" BEAKER_PUPPET_COLLECTION="puppet5" BEAKER_yum_full="yes"
```
Comment (ghoneycutt) [inline .travis.yml]: can we get rid of the other centos-7 test then?

File: spec/acceptance/99_client_server_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      package { 'cowsay':
+        ensure  => 'installed',
+        require => Yum::Repo['test'],
+      }
+      EOS
+      apply_manifest_on(client, pp, :catch_failures => true)
+      apply_manifest_on(client, pp, :catch_changes  => true)
+    end
+
+    describe yumrepo('test'), :node => client  do
```
Comment (ghoneycutt) [inline spec/acceptance/99_client_server_spec.rb]: from a style perspective, should all the tests that run on the client just be inside of an `it` block that has `on client` ?

File: .travis.yml (hunk 1 above)
Comment (treydock) [reply .travis.yml]: Actually used wrong set , updated PR to use correct new set that contains multiple nodes.

File: spec/acceptance/99_client_server_spec.rb (hunk 2 above)
Comment (treydock) [reply spec/acceptance/99_client_server_spec.rb]: I don't think that will work.  The `on client` idiom is just for executing shell commands on a node.

```
on client, 'run command'
```

If I understand correctly this won't work:

```
on client do
  it 'should do something' do
     test...
  end
end
```

The block of `on` will allow accessing things like stdout

```
on client, 'some command' do
  puts stdout
end
```

The `on` syntax I don't think is intended to dictate tests, just run commands.  The only way I know to use serverspec tests with multiple nodes is the `describe <resource>, :node => client` type pattern.  For example docs for yumrepo: https://serverspec.org/resource_types.html#yumrepo.  I've been using this pattern so long I'm not even sure where I found out about `:node => `, can't seem to find in serverspec advanced docs.

Upstream docs for `on`: https://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Helpers/HostHelpers#on-instance_method

**Comment by ghoneycutt** [comment]
Tests pass! Could you please squash/rebase.

**Comment by treydock** [comment]
Rebased and squashed.

**Comment by ghoneycutt** [comment]
Thank you!

**Comment by ghoneycutt** [comment]
Released in v2.1.0

**Pull request merged**